py/server/deephaven/multijoin.py

```diff
@@ -0,0 +1,126 @@
+#
```

Comment:
@jmao-denver What are your thoughts on where this stuff should live?  I can make arguments that it too should be in `deephaven.table`, since that is where the other joins are.  Not sure.

Reply:
Good question!
1. Based on the precedents of `PartitionedTable`, `TreeTable`, `RollupTable`, it does seem to make sense to place MultiJoinTable in table.py
2. I am not sure the level of 'table-ness' MultiJoinTable is. My guess is that it is at the same level as the above mentioned ones. The only difference is that those are created by methods on Table, not unbound functions. In a way, it is more similar to `table_factory.merge()`. 
3. One of the reasons we put `PartitionedTable` etc. in table.py is to avoid cyclical imports. MultiJoinTable doesn't seem to have this problem.

Although a bit torn, my personal preference is for consistency by having MultiJoinTable in table.py.  Maybe we can move the function `multi_join' to `table_factory`?

Reply:
I have followed these suggestions from @jmao-denver:

- `multi_join()` is moved to `table_factory.py`
- `MultiJoinTable`/`MultiJoinInput` are moved into `table.py`
- `multijoin.py` is removed

Reply:
- Also added `multi_join()` to `deephaven/__init__.py`